### PR TITLE
Makes the Quartermaster a Head of Staff

### PR DESCRIPTION
**Pull request by ItsMarmite**
image

## What Does This PR Do
In summary this PR separates Supply from the Head of Personnel, changing the Head of Personnels' required departmental hours to Service instead of Command and instates the Quartermaster as a member of Command, subject to SOP rewrite and gives them a new HUD. 

The Quartermaster's loadout has been tweaked to include a telebaton, mining shoes, a cigar and a new headset. The contents of the Quartermaster's locker has been updated to include both Cargo and Mining equipment, notably an extra Rapid Crate Sender and a Fulton Beacon/Pack for recovery of dead miners. 

Alongside this, it adds a mug for consistency with other Command roles for the Quartermaster.

The Quartermaster has had their access tweaked accordingly, with typical Command accesses of the brig, the bridge and so on being added to their ID card; alongside the Vault, which their remote has access to; consequentially the QM can no longer roll the nuke core or vault document theft objectives if they have rolled an antagonist roll.

Besides the aforementioned, various tweaks were made in the background to allow the QM to also be considered a Head via code for bans and the Revolution gamemode.

## Why It's Good For The Game

image

Look, I know this may be very, very controversial for some people but take a moment to hear me out before you bring out the pitchforks.  The current state of Supply and Service as Departments is pretty poor on the Command front; the HOP is overloaded with the amount of tasks they have to oversee when it comes to their departments, alongside the HOP line; leaving a QM that very often has the responsibilities of a full member of Command, without any of the oversight or behavioral standards we expect from a Command member. 

Through both this PR, and #22107 I hope to push forward to allowing both the roles of Quartermaster and Head of Personnel to specialise in their own departments and actually coordinate.

Currently if there' any failing in either department, the Quartermaster and the Head of Personnel are able to point fingers at eachother and lay responsibility of the department's failings on eachother, even when there are xeno nukie blob spiders gnawing on their shins.

Its hoped through this PR that not only will the Quartermaster have more responsibility, and therefore more encouragement to participate and push their department forward, but that the Head of Personnel will be able to perform similar duties with Service.

## Images of changes
image
image
image
image

## Testing
Spawning as the HOP and QM, testing access changes to both ID card and door remote changes. Checking loadout change, checking TGUI changes to crew PDAs, checking memory changes regarding departmental account information, so on, so on- there was a lot of me reloading the game and spawning as the Quartermaster. I'll likely update this section tomorrow with a cleaner list of showing everything as working.

## Changelog
:cl:
add: Adds Supply Management Console, mapped into Quartermaster's office in each map. 
add: The Quartermaster now has a mug, mapped into their office.
add: The Quartermaster now has their own headset and encryption key, added to loadout of Quartermaster.  
tweak: The Quartermaster is now a member of Command.
tweak: Changes roundstart tip regarding QM.
tweak: Mining telecommunications now required QM access to enter.
tweak: Changes Quartermaster roundstart locker contents/garment bag contents.
tweak: Supply Door Remote can now interact with station vault.
tweak: Alters HOP access to reflect separation between HOP and Supply Department.
tweak:  Death Alarm kit to include a death alarm implanter, instead of a normal implanter. 
tweak: Removes QM's ability to roll Nuke Core/Vault Document theft objectives.
tweak: Head of Personnel now requires Service hours instead of Command hours to unlock.
imageadd: Adds new HUD for Quartermaster role.
imagedel: Deletes old HUD for Quartermaster role.
/:cl:

### CODE REVIEW

File: code/game/machinery/suit_storage_unit.dm
Comment (Contrabang) [inline code/game/machinery/suit_storage_unit.dm:113]:
```suggestion
	mask_type = /obj/item/clothing/mask/gas/explorer
	req_access = list(ACCESS_QM)
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

**Comment by Contrabang** [comment]
Also, can you make your CL to have seperate lines for all the adds and tweaks? They should be on separate lines, them being too long may be a problem for the changelog because it may still have the length limit for a single "add:"

File: code/game/jobs/access.dm
Comment (GDNgit) [reply code/game/jobs/access.dm:167]: pretty sure marm gave it to them on request of denth so they could access the credits inside for biohazards/nukies

File: tgui/packages/tgui/interfaces/common/CrewManifest.js
Comment (GDNgit) [reply tgui/packages/tgui/interfaces/common/CrewManifest.js:33]: This does not work

File: _maps/map_files/Delta/delta.dmm
Comment (GDNgit) [reply _maps/map_files/Delta/delta.dmm:14870]: catexplode

mapping changes were kept to a minimum because of this

File: tgui/packages/tgui/interfaces/common/CrewManifest.js
Comment (Contrabang) [reply tgui/packages/tgui/interfaces/common/CrewManifest.js:33]: Really? Could you add back the false at least below it then? I trust javascript less with natively returning false.

File: code/game/jobs/access.dm
Comment (ItsMarmite) [reply code/game/jobs/access.dm:167]: Going to handle the other conversation tomorrow, but I gave the QM vault access after a discussion with Denth where we spoke about giving the vault a purpose other than nuke storage involving the QM. Until that idea is fleshed out- it makes sense to me for the QM to be entrusted with access to the station's money.

File: code/game/machinery/computer/card.dm
Comment (ItsMarmite) [reply code/game/machinery/computer/card.dm:762]: I've reached out to spriting to see if a different icon could be made for the console.

**Comment by ArcticValor** [comment]
going to put it in now, "ALL HAIL CARGONIA!" thank you.

**Comment by SynthTwo** [comment]
LETS FUCKING GOOOO!!!

**Comment by Mitchs98** [comment]
When/if this is merged I will become a QM main. o7

File: code/game/objects/items/weapons/storage/boxes.dm
Comment (warriorstar-orion) [inline code/game/objects/items/weapons/storage/boxes.dm:315]: I was going to check if the BS got another implant for the QM and noticed this change, is the intention to spawn the implanter with one death alarm in it rather than add another death alarm implant case? if so that may merit a mention in the CL to head off questions by confused blueshields

also
```suggestion
	new /obj/item/implanter/death_alarm(src)
```

Comment (ItsMarmite) [reply code/game/objects/items/weapons/storage/boxes.dm:315]: I'll clarify this in the changelog. It was intentional to spawn a death alarm implanter instead of another implant case, mostly to avoid further bag clutter honestly; I could always change it back to an implanter and another death alarm case if you'd like?

**Comment by ItsMarmite** [comment]
> Also, can you make your CL to have seperate lines for all the adds and tweaks? They should be on separate lines, them being too long may be a problem for the changelog because it may still have the length limit for a single "add:"

I believe this is what you meant? Cleaned up the changelog a little, let me know if there's still any issues.

File: code/game/machinery/computer/card.dm
Comment (ItsMarmite) [reply code/game/machinery/computer/card.dm]: image

I am no spriter, but there's my best attempt, now with a new icon screen under idsup.

File: code/game/objects/items/weapons/storage/boxes.dm
Comment (warriorstar-orion) [reply code/game/objects/items/weapons/storage/boxes.dm:315]: I don't have a strong preference either way, just wanted to make sure I understood the change.

**Comment by Contrabang** [comment]
> > Also, can you make your CL to have seperate lines for all the adds and tweaks? They should be on separate lines, them being too long may be a problem for the changelog because it may still have the length limit for a single "add:"
> 
> I believe this is what you meant? Cleaned up the changelog a little, let me know if there's still any issues.

Yep, CL looks good to me

File: _maps/map_files/Delta/delta.dmm
Comment (ItsMarmite) [reply _maps/map_files/Delta/delta.dmm:14870]: In fairness it touches nothing in relation to the power grid and jut adds some console/machines while changing some door access.

File: code/game/objects/structures/crates_lockers/closets/secure/cargo_lockers.dm
Comment (matttheficus) [inline code/game/objects/structures/crates_lockers/closets/secure/cargo_lockers.dm:35]: Needs a flash like the other Heads.

**Review by warriorstar-orion — CHANGES_REQUESTED** [review]
Code changes are fine other than review comments.
Map changes are fine. We absolutely need to enlarge most QM offices at some point. 

As I mentioned in Discord I think this is a constructive change and I hope it gets approval to move forward. As someone who mains cargo, service, and command, setting expectations of the QM to be a head role, and removing much of the HOP's current duties so that they can focus on service, should really be beneficial to both supply and service in the long term. I can't really speak with any authority for how it might affect antag gameplay/balance, however.

File: code/game/machinery/computer/buildandrepair.dm
Comment (Henri215) [inline code/game/machinery/computer/buildandrepair.dm:226]: Why not just do a ```board_name = "QM ID Computer"``` like the other ID boards?

File: code/game/machinery/computer/card.dm
Comment (Henri215) [inline code/game/machinery/computer/card.dm:763]: Light color red? Is this intended?

File: code/game/objects/structures/crates_lockers/closets/secure/cargo_lockers.dm
Comment (Henri215) [inline code/game/objects/structures/crates_lockers/closets/secure/cargo_lockers.dm:32]: They are supposed to go down on lavaland now?

**Review by Henri215 — COMMENTED** [review]
You will have to add QM to the ```blacklisted_jobs``` list inside the ```code/modules/events/bureaucratic_error.dm```, since the HoP cant change command job slots.

File: code/game/objects/structures/crates_lockers/closets/secure/cargo_lockers.dm
Comment (ItsMarmite) [reply code/game/objects/structures/crates_lockers/closets/secure/cargo_lockers.dm:32]: The intention is for them to be able to assist more easily in recovery of dead miners, if they wish. Half of their job revolves around coordinating and ensuring mining is able to function. This just gives them some additional tools for getting to, and recovering deceased minerss.

File: code/game/machinery/computer/card.dm
Comment (ItsMarmite) [reply code/game/machinery/computer/card.dm:763]: As far as I can tell the light color doesn't actually do anything or change anything with this console, but I'll check and get back to you.

File: code/game/machinery/computer/buildandrepair.dm
Comment (ItsMarmite) [reply code/game/machinery/computer/buildandrepair.dm:226]: This'll be due to me having used an old attempt to make QM a head as a reference, seems I forgot to change this when I was making the first changes. This'll be updated in the next commit.

File: code/game/jobs/departments.dm
Comment (ItsMarmite) [reply code/game/jobs/departments.dm:126]: Woops, will be fixed in newest commit

File: code/game/machinery/computer/card.dm
Comment (ItsMarmite) [reply code/game/machinery/computer/card.dm:763]: So after some testing, the light_color var doesn't seem to really matter at all, and is leftover from an old PR. See the following images. One is with me varediting the light range to the value of 10, up from its current value. 

image

As you can see, it doesn't produce any red light until the range is VV edited.

image

In which case it has a very dull red colour to it.

File: code/game/jobs/job/supervisor.dm
Comment (ItsMarmite) [reply code/game/jobs/job/supervisor.dm:76]: You're right there, this has been changed in an upcoming commit push.

File: code/game/objects/items/weapons/storage/garment.dm
Comment (ItsMarmite) [reply code/game/objects/items/weapons/storage/garment.dm:207]: Not too sure what you mean by boxes.dm but I've fixed the accidental space in a new commit.

File: code/game/objects/structures/crates_lockers/closets/secure/cargo_lockers.dm
Comment (ItsMarmite) [reply code/game/objects/structures/crates_lockers/closets/secure/cargo_lockers.dm:35]: Been added to the locker.

File: code/game/machinery/computer/card.dm
Comment (Henri215) [reply code/game/machinery/computer/card.dm:763]: I would still change it to yellow just in case

File: code/game/machinery/computer/buildandrepair.dm
Comment (Henri215) [inline code/game/machinery/computer/buildandrepair.dm:226]:
```suggestion
	board_name = "Supply ID Computer"
```
Forgot that

File: code/game/machinery/computer/card.dm
Comment (ItsMarmite) [reply code/game/machinery/computer/card.dm:763]: image

Changed to a more fitting colour in the next commit. (This is the console with the light range and power upped way high)

File: code/modules/events/bureaucratic_error.dm
Comment (Henri215) [inline code/modules/events/bureaucratic_error.dm:21]:
```suggestion
		/datum/job/warden,
		/datum/job/qm
	)
```
:)

**Comment by OctusGit** [comment]
speedmerge this shit immediately

File: code/game/objects/structures/crates_lockers/closets/secure/cargo_lockers.dm
Comment (ItsMarmite) [reply code/game/objects/structures/crates_lockers/closets/secure/cargo_lockers.dm]: After some discussion with Henri in coding chat, plasma cutter has been removed, with a GPS and crew monitor handed to the QM locker. The QM should be coordinating search and rescues FROM the mining base, not running into the wastes himself.

**Comment by Henri215** [comment]
Are you going to close https://github.com/ParadiseSS13/Paradise/pull/22107 and tweak the time req in this PR once it gets approval? Because you would need the requirement time before making the QM a head of staff.

**Comment by procdrone** [comment]
very interesting

**Comment by Cress67** [comment]
We'll finally get to see QMs maybe actually try to prevent Cargo from becoming Revs rather than immediately selling out, I fully approve.

**Comment by ItsMarmite** [comment]
> Are you going to close #22107 and tweak the time req in this PR once it gets approval? Because you would need the requirement time before making the QM a head of staff.

Given the current map freeze, I'll be keeping #22107 open. I don't see this PR being merged, if approved, for a good while yet. If fully approved #22107 will be a good transition period.

**Comment by S34NW** [comment]
PR currently frozen/on hold due to map edits and the mapping freeze caused by https://github.com/ParadiseSS13/Paradise/pull/21250

**Comment by S34NW** [comment]
Map freeze has been lifted, PR no longer on hold.

**Comment by Pythbit** [comment]
Hail Cargonia

File: code/game/jobs/job/support.dm
Comment (Henri215) [inline code/game/jobs/job/support.dm:9]:
```suggestion
	supervisors = "the captain"
```
Following the way it is written over other jobs

**Review by Henri215 — APPROVED** [review]
Mapping and sprites are looking good to me.

File: code/game/objects/items/weapons/storage/garment.dm
Comment (Henri215) [inline code/game/objects/items/weapons/storage/garment.dm:203]: Going to keep the explorer suit?

Loader modsuits allows people to survive in lavaland? I dont know much about modsuits.

**Comment by ArcticValor** [comment]
Just thought about this just now but will the blueshield come in with 7 death implants due to there now being 7 heads? (HOP, HOS, CMO, Captain, CE, RD, and now the QM?)

**Comment by Henri215** [comment]
> Just thought about this just now but will the blueshield come in with 7 death implants due to there now being 7 heads? (HOP, HOS, CMO, Captain, CE, RD, and now the QM?)

Yes, he did update the number of implants.

**Review by Contrabang — CHANGES_REQUESTED** [review]
/datum/game_mode/revolution/announce() will need to be updated to include the QM in the announcement

File: code/game/objects/items/weapons/storage/garment.dm
Comment (ItsMarmite) [reply code/game/objects/items/weapons/storage/garment.dm:203]: Nah the loader is definitely more of a 'delivering crates' than Lavaland thing. I threw the suit in for the occasional lowpop scenario of the QM having to do a small bit of mining, or occasional rescue.

File: strings/sillytips.txt
Comment (Contrabang) [inline strings/sillytips.txt:32]: Whats going on here, some kind of weird end of line change?

**Review by Contrabang — APPROVED** [review]
LGTM except this one minor thing

**Review by S34NW — COMMENTED** [review]
Fix meta CE office pixel shift on keycard and we're good to go

File: strings/sillytips.txt
Comment (S34NW) [reply strings/sillytips.txt:32]: probably a change to CRLF

Comment (GDNgit) [reply strings/sillytips.txt:32]: no this is a newline at the end of the file, shouldn't be there, i'll fix it

**Review by warriorstar-orion — APPROVED** [review]
LGTM

**Review by S34NW — APPROVED** [review]
LETS GO TIME FOR QM HEAD

**Comment by Contrabang** [comment]
HOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOLY SHIT!

**Pull request merged**
